lib/string_reverse.rb

```diff
@@ -1,6 +1,23 @@
 # A method to reverse a string in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: linear, O(n)
+#                  where n is the length of my_string
+# Space complexity: O(1)
+#                   where n is the length of my_string * the number of bytes a
+#                   character requires for space
 def string_reverse(my_string)
-  raise NotImplementedError
+  return nil if my_string == nil
+
+  i = 0
+  j = my_string.length - 1
+  temp = ""
+
+  while i < j
+    temp = my_string[i]

```

Comment:
Be sure to test for those pesky "off-by-one" errors! :smile:

It's outside the scope of this exercise - but what are some test cases you might use to check for them?